Solution1/ConsoleApp/Services/UserCommunicateService.cs

```diff
@@ -135,5 +143,62 @@ private async Task GetForecastByCityNameAsync()
             var result = await _invoker.RunAsync(command);
             Console.WriteLine(result.GetMultiStringRepresentation());
         }
+
+        private async Task GetBestWeatherByArrayCityNameAsync()
+        {
+            Console.WriteLine("Please, enter array city name (separator symbal - ',') :");
+            var arrayCityNames = Console.ReadLine();
+            if (string.IsNullOrEmpty(arrayCityNames))
+            {
+                Console.WriteLine(Constants.Validation.IncorrectValue);
+                _logger.LogError($"User entered incorrect value for 'PeriodOfDays'.");
+                return;
+            }
+
+            var command = new BestWeatherCommand(_weatherServiсe, arrayCityNames);
+            var dictionaryWeatherResponsesDTO = await _invoker.RunAsync(command);
+
+            var countSuccessResponse = dictionaryWeatherResponsesDTO.TryGetValue(true, out var successfulWeatherResponses) ? successfulWeatherResponses.Count : 0;
+            var countFailResponse = dictionaryWeatherResponsesDTO.TryGetValue(false, out var failedWeatherResponses) ? failedWeatherResponses.Count : 0;
+
+            if (countSuccessResponse > 0)
+            {
+                var bestWeather = successfulWeatherResponses.OrderByDescending(w => w.Temp).FirstOrDefault();
+                Console.WriteLine($"City with the highest temperature {bestWeather.Temp} C: {bestWeather.CityName}. " +
+                    $"Successful request count: {countSuccessResponse}, failed: {countFailResponse}.");
+            }
+            else
+            {
+                Console.WriteLine($"Error, no successful requests. Failed requests count: {countFailResponse}");
+            }
+
+            if (Convert.ToBoolean(ConfigurationManager.AppSettings["isDebugMode"]))
+            {
+                ShowDebugInformation(successfulWeatherResponses, failedWeatherResponses);
+            }
+
+            return;
+        }
+
+        private void ShowDebugInformation(IEnumerable<WeatherResponseDTO> successfulWeatherResponses, IEnumerable<WeatherResponseDTO> failedWeatherResponses)
```

Comment:
print cities in the order they were in the result, not success in the begging failed in the end

Reply:
+